eth/api_tracer.go

```diff
@@ -719,6 +719,30 @@ func (api *PrivateDebugAPI) TraceTransaction(ctx context.Context, hash common.Ha
 	return api.traceTx(ctx, msg, vmctx, statedb, config)
 }
 
+// Call lets you trace a given eth_call. returns the structured logs created during the execution of EVM
+// if the given transaction was added on top of the provided block and returns them as a JSON object.
+// You can provide -2 as a block number to trace on top of the pending block.
+func (api *PrivateDebugAPI) TraceCall(ctx context.Context, args ethapi.CallArgs, blockNrOrHash rpc.BlockNumberOrHash, config *TraceConfig) (interface{}, error) {
+	// First try to retrieve the state
+	statedb, header, err := api.eth.APIBackend.StateAndHeaderByNumberOrHash(ctx, blockNrOrHash)
+	if err != nil {
+		// try to recompute the state
+		reexec := defaultTraceReexec
+		if config != nil && config.Reexec != nil {
+			reexec = *config.Reexec
+		}
+		_, _, statedb, err = api.computeTxEnv(*blockNrOrHash.BlockHash, 0, reexec)
```

Comment:
If the `blockNrOrHash ` only specifies the `blockNumber`, the hash may be nil.

Reply:
Yeah, I can repro the panic with nil pointer, please fix it. Otherwise LGTM

Reply:
Thank you for finding that! I fixed it, could you maybe share your repro steps, so that I can test it a bit further?

Reply:
yeah, unfortunately, I test it manually by specifying an old block number. But it will be nice if we can have the unit test for it.

Btw, can we retrieve the relevant block hash if there is the only number specified?